nvbench/cuda_stream.cuh

```diff
@@ -42,10 +45,18 @@ struct cuda_stream
    * Constructs a cuda_stream that owns a new stream, created with
    * `cudaStreamCreate`.
    */
-  cuda_stream()
-      : m_stream{[]() {
+  cuda_stream(std::optional<nvbench::device_info> device)
```

Comment:
Docs should be updated to explain the semantics of the new `device` parameter.

Reply:
Thanks. Updated docs. Could you please check if it's understandable?